cmake/OpenCVUtils.cmake

```diff
@@ -1763,15 +1763,15 @@ endif()
 
 macro(ocv_git_describe var_name path)
   if(GIT_FOUND)
-    execute_process(COMMAND "${GIT_EXECUTABLE}" describe --tags --exact-match --dirty
+    execute_process(COMMAND "${GIT_EXECUTABLE}" --work-tree="${path}" describe --tags --exact-match --dirty
```

Comment:
"Rooting" is not expected and breaks opencv_contrib at least:

```
--   Extra modules:
--     Location (extra):            /build/precommit_linux64/opencv_contrib/modules
--     Version control (extra):     unknown
```

Consider passing `OPENCV_VCSVERSION` value via CMake (command line or upstream scripts).